examples/hello_world/client/hello_world.cc

```diff
@@ -38,6 +38,11 @@ void say_hello(HelloWorld::Stub* stub, std::string name) {
   if (!status.ok()) {
     LOG(WARNING) << "Could not call SayHello('" << name << "'): " << status.error_code() << ": "
                  << status.error_message();
+    // The server code includes a case with a deliberate error to check gRPC
+    // error code propagation; any other error should cause the client to fail.
+    if (status.error_code() != grpc::FAILED_PRECONDITION) {
+      LOG(QFATAL) << "Error code is not the deliberately-induced error!";
+    }
```

Comment:
I feel this example is trying too hard to be both an interactive client and a test case. Maybe we should have a `--test` flag similar to one of the other examples? So that it may expect the error only on the appropriate query. Or have two clients, one interactive and one "batch"?

Reply:
Good point, raised #639 to cover moving platform-testing functionality out of this example.

Reply:
Thanks, feel free to submit this in the meanwhile if you prefer.